File: PartTwo/Social/ViewController.m
//
//  ViewController.m
//  Social
//
//  Created by Varindra Hart on 9/18/15.
//  Copyright © 2015 Varindra Hart. All rights reserved.
//

#import "ViewController.h"
#import "CustomView.h"

@interface ViewController () <CustomViewProtocol>
@property (weak, nonatomic) IBOutlet UIView *backgroundView;
@property (weak, nonatomic) IBOutlet UIView *customViewContainer;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSArray *views = [[NSBundle mainBundle]loadNibNamed:@"View" owner:self options:nil];
    CustomView *customView = [views firstObject];
    customView.delegate = self;
    
    [self.customViewContainer addSubview:customView];
    customView.frame = self.customViewContainer.bounds;
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma CustomViewProtocol Methods

- (void)socialMediaViewDidTapCommentButton:(CustomView *)view{
    [self.backgroundView setBackgroundColor:[UIColor greenColor]];
}

- (void)socialMediaViewDidTapLikeButton:(CustomView *)view{
    self.backgroundView.backgroundColor = [UIColor blueColor];
}

- (void)socialMediaViewDidTapShareButton:(CustomView *)view{
    self.backgroundView.backgroundColor = [UIColor redColor];
}
@end

File: Social/Social/ViewController.swift
//
//  ViewController.swift
//  Social
//
//  Created by Jovanny Espinal on 9/14/15.
//  Copyright © 2015 Jovanny Espinal. All rights reserved.
//

import UIKit

class ViewController: UIViewController, SocialDelegate {

    @IBOutlet weak var socialMediaView: SocialMediaView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        socialMediaView.delegate = self
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func socialMediaViewDidTapCommentButton(view: SocialMediaView) {
        self.view.backgroundColor = UIColor.greenColor()
    }
    
    func socialMediaViewDidTapLikeButton(view: SocialMediaView) {
        self.view.backgroundColor = UIColor.blueColor()
    }
    
    func socialMediaViewDidTapShareButton(view: SocialMediaView) {
        self.view.backgroundColor = UIColor.redColor()
    }
    

}

